sections/main-product.liquid

```diff
@@ -49,23 +49,36 @@
       {% if product_system != nil %}› <a href="/collections/{{ product_system | downcase | replace: "/", "" | replace: "  ", " " | replace: " ", "-" }}" class="link">{{ product_system }}</a>{% endif %}
     </div>
   {% endif %}
-  <div class="product grid grid--1-col {% if product.media.size > 0 %}grid--2-col-tablet{% else %}product--no-media{% endif %}">
+  <div class="product grid grid--1-col grid--2-col-tablet">
     <div class="grid__item product__media-wrapper">
       <slider-component class="slider-mobile-gutter">
         <a class="skip-to-content-link button visually-hidden" href="#ProductInfo-{{ section.id }}">
           {{ "accessibility.skip_to_product_info" | t }}
         </a>
         <ul class="product__media-list grid grid--peek list-unstyled slider slider--mobile" role="list">
-          {%- assign variant_images = product.images | where: 'attached_to_variant?', true | map: 'src' -%}
+          {%- if product.media.size > 0 -%}
+            {%- assign variant_images = product.images | where: 'attached_to_variant?', true | map: 'src' -%}
+          {%- else -%}
+          {%- if product.metafields.kameratori.host != blank -%}
+            {%- assign host_product = product.metafields.kameratori.host.value -%}
+            {%- assign variant_images = host_product.images | where: 'attached_to_variant?', true | map: 'src' -%}
+            {%- endif -%}
+          {%- endif -%}
           {%- if product.selected_or_first_available_variant.featured_media != null -%}
             {%- assign media = product.selected_or_first_available_variant.featured_media -%}
             <li class="product__media-item grid__item slider__slide{% if media.media_type != 'image' %} product__media-item--full{% endif %}{% if section.settings.hide_variants and variant_images contains media.src %} product__media-item--variant{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
               {% render 'product-thumbnail', media: media, position: 'featured', loop: section.settings.enable_video_looping, modal_id: section.id, xr_button: true %}
             </li>
           {%- endif -%}
+          {%- if host_product.featured_media != null-%}
```

Comment:
Also, maybe it makes sense to use **elsif** to make sure that only one of these conditions can ever run so this doesn't directly depend on previous variables being set in a specific way. 
I mean I realize that the second branch can only be accessed in the case that the first is because the `host_product` variable isn't initialized in the case where the first branch is true, but this is indirect logic which is hard to read and can break easily if the previous part of code is modified, for example by moving the `host_product` definition out of the condition, which might be a logical thing to do under some circumstances

Reply:
It makes total sense, implemented it.